cpp/src/arrow/compute/exec/ir_consumer.cc

```diff
@@ -625,7 +625,7 @@ Result<Declaration> Convert(const ir::Relation& rel) {
             break;
         }
 
-        NullPlacement key_null_placement;
+        NullPlacement key_null_placement{};
```

Comment:
Perhaps we could add a default case to the switch statement instead?  Although I think this code is going away so probably not worth it to nitpick.